sklearn/linear_model/base.py

```diff
@@ -523,7 +523,7 @@ def _pre_fit(X, y, Xy, precompute, normalize, fit_intercept, copy):
         precompute = False
         X, y, X_offset, y_offset, X_scale = _preprocess_data(
             X, y, fit_intercept=fit_intercept, normalize=normalize,
-            return_mean=True)
+            copy=copy, return_mean=True)
```

Comment:
I suspect you should never copy for a sparse matrix as X will not be densified hence modified anyway. So copy should be False all the time. Can you check?